ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/translator/ASTConverter.java

```diff
@@ -218,6 +218,13 @@ private static ASTNode createNullField(RelDataType fieldType) {
       return mapCallNode.node();
     }
 
+    if (fieldType.getSqlTypeName() == SqlTypeName.ARRAY) {
```

Comment:
What about other complex types .. STRUCT and MAP ? 
e.g  create table t1 (c3 struct <name: string, age: int>);
       CREATE TABLE t2 (c1 int, c2 map<c2:string, c3:int>);
A struct can also appear within an array but that might be handled by this code change.

Reply:
those are handled above this change:
https://github.com/apache/hive/blob/53cb10347bec803008b17ff05c48536fe9e722e9/ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/translator/ASTConverter.java#L203-L219

Reply:
Ah, missed that.  Can we add  q tests for struct and map as well ?